tests/test_updater_ng.py

```diff
@@ -109,14 +113,96 @@ def tearDown(self):
+    )
+    shutil.copy(source_path, destination_path)
+
+
+  def _root_make_consistent_snapshot_true(self) -> bytes:
+    """Instantiate a root Metadata instanec, change roots consistent_snapshot to
+    true, self sign the changed root and then serialize the modified root back
+    to bytes."""
+    root_path = os.path.join(self.repository_directory, "metadata", "root.json")
+    root = Metadata.from_file(root_path)
+    root.signed.consistent_snapshot = True
+    root_key_path = os.path.join(self.keystore_directory, "root" + '_key')
+    root_key_dict = import_rsa_privatekey_from_file(
+            root_key_path, password="password"
+    )
+    root_signer = SSlibSigner(root_key_dict)
+    root.sign(root_signer)
+    return root.to_bytes()
+
+
+  def test_refresh_on_consistent_targets(self):
+    # Test refresh with consistent snapshot.
+
+    # Create consistent targets with file path HASH.FILENAME.EXT
+    self._create_consistent_target("targets.json", "file1.txt")
+    self._create_consistent_target("role1.json", "file3.txt")
+
+    # Make root consistent snapshot true.
+    root_modified_bytes = self._root_make_consistent_snapshot_true()
+    # Instantiate a new TrustedMetadataSet with changed root and modify internals.
+    # This seems better than modifying the actual root file in the client directory
+    # considering that the client directory is used from the rest of the tests.
+    trusted_metadata_set = TrustedMetadataSet(root_modified_bytes)
+    self.repository_updater._trusted_set = trusted_metadata_set
+
+    # All metadata is in local directory already
+    self.repository_updater.refresh()
+
+    # Get targetinfo for 'file1.txt' listed in targets
+    targetinfo1 = self.repository_updater.get_one_valid_targetinfo("file1.txt")
+    # Get targetinfo for 'file3.txt' listed in the delegated role1
+    targetinfo3= self.repository_updater.get_one_valid_targetinfo("file3.txt")
+
+    destination_directory = self.make_temp_directory()
+    updated_targets = self.repository_updater.updated_targets([targetinfo1, targetinfo3],
+                                                      destination_directory)
+
+    self.assertListEqual(updated_targets, [targetinfo1, targetinfo3])
+    self.repository_updater.download_target(targetinfo1, destination_directory)
+    updated_targets = self.repository_updater.updated_targets(updated_targets,
+                                                      destination_directory)
+
+    self.assertListEqual(updated_targets, [targetinfo3])
+
+
+    self.repository_updater.download_target(targetinfo3, destination_directory)
+    updated_targets = self.repository_updater.updated_targets(updated_targets,
+                                                  destination_directory)
+
+    self.assertListEqual(updated_targets, [])
+
   def test_refresh(self):
+    # Test refresh without consistent snapshot.
+    self.repository_updater.config.prefix_targets_with_hash = False
```

Comment:
This config shouldn't be doing anything without consistent_snapshot being set. I'm not sure this explicit disable is needed.